src/store.c

```diff
@@ -351,14 +352,18 @@ int pthreads_store_write(zval *object, zval *key, zval *write) {
 				normal refcounting, we'll just never read the reference
 			*/
 			rebuild_object_properties(&threaded->std);
+			
+			pthreads_store_sync(object);
```

Comment:
A test case might be difficult. I ran the script above for a certain time and watched the memory consumption. Any idea to trace this in a test script? 

I've added pthreads_store_sync at this point to guarantee consistency with the behavior of reading properties.